Cocoaheads School_Lesson2.playground/Contents.swift

```diff
@@ -19,31 +19,51 @@ class Stack {
 }
 
 class StackStatistics: Stack {
-    private var minValues = [Int]()
-
+    private var minElement: Int?
+    
     override func push(element: Int) {
-        super.push(element: element)
-        if let minValue = minValues.last {
-            if minValue > element {
-                minValues.append(element)
+        if let minElement = minElement {
+            if minElement > element {
+                super.push(element: (2 * element - minElement))
```

Comment:
In general, I'd recommend to make it with 2 lines.

```suggestion
                let elementThatStoresInformationAboutMinimum = 2 * element - minElement
                super.push(element: elementThatStoresInformationAboutMinimum)
```

It allows us to make it more readable and we can describe what's the purpose of `2 * element - minElement`.
Does that make sense?

Reply:
About the trailing spaces: 
I'm trying to delete it, but they appear again.

> _In general, I'd recommend to make it with 2 lines._
 Thank you for this recommend. Yes, it's make the code more readable. Done

Reply:
_I'm still able to see the trailing spaces._ - Done